### PR TITLE
Add support for building against GEOS development versions

### PR DESCRIPTION
**Pull request by strk**
In these cases the GEOS_VERSION_PATCH ends with a non-numeric
element, like "2dev". This patch extract an integer from it.

### CODE REVIEW

**Comment by jef-n** [comment]
Um, why do we need a geosVersionPatch() when there is geosVersionInt()?

**Comment by strk** [comment]
geosVersionInt also needed a fix, which I now pushed. Why we need geosVersionPatch() I don't know, maybe we don't need it, but I guess by policy we don't remove public methods ?

File: src/core/qgis.cpp
Comment (m-kuhn) [inline src/core/qgis.cpp:36]: This would have been THE opportunity to call this `strk`

Comment (strk) [reply src/core/qgis.cpp:36]: xstr is a common name for such macro, if you search around :)
Do I need approval on the PR to merge ? Lost track of policy changes, if I do, could you please take the opportunity to approve it ? :)

File: src/core/qgis.cpp
Comment (m-kuhn) [inline src/core/qgis.cpp:316]:
```suggestion
  static const int version = QStringLiteral( "%1%2%3" )
         .arg( GEOS_VERSION_MAJOR, 2, 10, QChar( '0' ) )
         .arg( GEOS_VERSION_MINOR, 2, 10, QChar( '0' ) )
         .arg( geosVersionPatch(), 2, 10, QChar( '0' ) ).toInt()
         ;
         return version;
```

File: src/core/qgis.cpp
Comment (m-kuhn) [reply src/core/qgis.cpp:36]: Yep, you need, will do as soon as I get a reply or patch for the other comment ;-)

File: src/core/qgis.cpp
Comment (strk) [reply src/core/qgis.cpp:316]: Indentation was made by prepare_commit.sh, and passed CI style checks.
Do you really think that saving the per-process function call is worth the trouble of finding a way to to numbers-from-string extraction in the preprocessor ? I'd be happy with whawt we have here

Comment (strk) [reply src/core/qgis.cpp:316]: Ah, I see the suggested change, sorry I didn't notice before.
Ok, will do, and also for geosVersionPatch() -- pity, as CI finally passed :)

Comment (strk) [reply src/core/qgis.cpp]: Pushed. @m-kuhn can you see if you like it now ?

**Review by m-kuhn — APPROVED** [review]

**Comment by m-kuhn** [comment]
Looks good to me, thanks

**Comment by jef-n** [comment]
> geosVersionInt also needed a fix, which I now pushed. Why we need geosVersionPatch() I don't know, maybe we don't need it, but I guess by policy we don't remove public methods ?

It's a new method (introduced in 3.20, which doesn't exist yet) - unused.  Only geosVersionInt() is used, but only in tests to adapt checks for GEOS 3.9.

**Comment by strk** [comment]
> > geosVersionInt also needed a fix, which I now pushed. Why we need geosVersionPatch() I don't know, maybe we don't need it, but I guess by policy we don't remove public methods ?
> 
> It's a new method (introduced in 3.20, which doesn't exist yet) - unused. Only geosVersionInt() is used, but only in tests to adapt checks for GEOS 3.9.

Ok, then if you want I drop it and wait for the 4th CI to pass

**Comment by lbartoletti** [comment]
> It's a new method (introduced in 3.20, which doesn't exist yet) - unused. Only geosVersionInt() is used, but only in tests to adapt checks for GEOS 3.9.

@strk @jef-n Indeed I introduced these methods to adapt tests with GEOS 3.9 and to expose the version in python.  I don't think we need geosVersionPatch() a day, but it was for completeness.

**Pull request merged**
